index.js

```diff
@@ -8,92 +8,71 @@ const RedisStore = require('./store/RedisStore')
+}
 
-  const draftSpecHeaders = {
-    rateLimit: 'ratelimit-limit',
-    rateRemaining: 'ratelimit-remaining',
-    rateReset: 'ratelimit-reset',
-    retryAfter: 'retry-after'
-  }
+const draftSpecHeaders = {
+  rateLimit: 'ratelimit-limit',
+  rateRemaining: 'ratelimit-remaining',
+  rateReset: 'ratelimit-reset',
+  retryAfter: 'retry-after'
+}
 
-  // create the object that will hold the "main" settings that can be shared during the build
-  // 'global' will define, if the rate limit should be apply by default on all route. default : true
+async function fastifyRateLimit (fastify, settings) {
   const globalParams = {
     global: (typeof settings.global === 'boolean') ? settings.global : true
   }
 
   if (typeof settings.enableDraftSpec === 'boolean' && settings.enableDraftSpec) {
     globalParams.enableDraftSpec = true
-    labels = draftSpecHeaders
+    globalParams.labels = draftSpecHeaders
   } else {
     globalParams.enableDraftSpec = false
+    globalParams.labels = defaultHeaders
   }
 
   globalParams.addHeaders = Object.assign({
-    [labels.rateLimit]: true,
-    [labels.rateRemaining]: true,
-    [labels.rateReset]: true,
-    [labels.retryAfter]: true
+    [globalParams.labels.rateLimit]: true,
+    [globalParams.labels.rateRemaining]: true,
+    [globalParams.labels.rateReset]: true,
+    [globalParams.labels.retryAfter]: true
   }, settings.addHeaders)
 
   globalParams.addHeadersOnExceeding = Object.assign({
-    [labels.rateLimit]: true,
-    [labels.rateRemaining]: true,
-    [labels.rateReset]: true
+    [globalParams.labels.rateLimit]: true,
+    [globalParams.labels.rateRemaining]: true,
+    [globalParams.labels.rateReset]: true
   }, settings.addHeadersOnExceeding)
 
-  globalParams.labels = labels
-
-  // define the global maximum of request allowed
+  // Global maximum allowed requests
   globalParams.max = ((typeof settings.max === 'number' && !isNaN(settings.max)) || typeof settings.max === 'function')
     ? settings.max
     : 1000
 
-  // define the global Time Window
+  // Global time window
   globalParams.timeWindow = typeof settings.timeWindow === 'string'
     ? ms.parse(settings.timeWindow)
     : typeof settings.timeWindow === 'number' && !isNaN(settings.timeWindow)
       ? settings.timeWindow
       : 1000 * 60
-
-  globalParams.timeWindowInSeconds = (globalParams.timeWindow / 1000) | 0
+  globalParams.timeWindowInSeconds = Math.floor(globalParams.timeWindow / 1000)
 
   globalParams.hook = settings.hook || defaultHook
   globalParams.allowList = settings.allowList || settings.whitelist || null
-  globalParams.ban = settings.ban || null
-  globalParams.onBanReach = typeof settings.onBanReach === 'function' ? settings.onBanReach : undefined
-  globalParams.onExceeding = typeof settings.onExceeding === 'function' ? settings.onExceeding : undefined
-  globalParams.onExceeded = typeof settings.onExceeded === 'function' ? settings.onExceeded : undefined
+  globalParams.ban = settings.ban || 0
```

Comment:
I wonder what this means. 

If i set ban to 1 it means one 429 then you get banned. if i set it to 0 it means you get immediatly banned without seeing a 429. Is there a unit test for that?

Reply:
I was just coercing the type to Number since 0 is falsy and equivalent to `null`, `undefined`, or `false`

Ban only works if it is [truthy](https://github.com/fastify/fastify-rate-limit/blob/master/index.js#L253) (!==0), so 0 kind of turns it off in the same way `undefined` does

I find it more consistent to use fasly values and keep a variable type always the same when possible, but you raised a good point that we might in the future want to add a functionality where ban === 0 means a request is always banned, I will revert it

Reply:
I dont know if the current implementation did even take care of that.